gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -202,14 +202,11 @@ DatasetDescriptor getDestinationDataset() {
   private DatasetDescriptor getDatasetDescriptor(Optional<String> stringMetastoreURI) {
     String destinationTable = this.getDbName() + "." + this.getInputTableName();
 
-    URI hiveMetastoreURI = null;
-    if (stringMetastoreURI.isPresent()) {
-      hiveMetastoreURI = URI.create(stringMetastoreURI.get());
-    }
+    URI hiveMetastoreURI = stringMetastoreURI.isPresent() ? URI.create(stringMetastoreURI.get()) : null;
 
     DatasetDescriptor destinationDataset =
         new DatasetDescriptor(DatasetConstants.PLATFORM_ICEBERG, hiveMetastoreURI, destinationTable);
-    destinationDataset.addMetadata(DatasetConstants.FS_URI, this.getFs().getUri().toString());
+    destinationDataset.addMetadata(DatasetConstants.FS_URI, this.getSourceFs().getUri().toString());
```

Comment:
Why is the destinationDataset using the source FS? It can be on a different filesystem uri here

Reply:
updated the File System to be as per source and destination in my latest commit